scripts/benchmark.py

```diff
@@ -0,0 +1,238 @@
+
+
+def _run_benchmark_suite(resources_dir: Path):
+    # Default reader / imageio imread tests
+
+    # CZI reader / czifile imread tests
+    czi_reader_results = _run_benchmark(
+        resources_dir=resources_dir,
+        extensions=["*.czi"],
+        non_aics_reader=czifile.imread,
+    )
+
+    return [
+        *czi_reader_results,
+    ]
+
+
+def run_benchmarks(args: Args):
+    # Results are stored as they are returned
+    all_results = {}
+
+    # Try running the benchmarks
+    #try:
+    # Get benchmark resources dir
+    resources_dir = Path().resolve().parent / "aicspylibczi" / "tests" / "resources"
+    print(resources_dir.resolve())
+    # Store machine config
+    _ = {
+        "platform": platform.system(),
+        "platform_version": platform.version(),
+        "architecture": platform.machine(),
+        "cpu_total_count": psutil.cpu_count(),
+        "cpu_current_utilization": psutil.cpu_percent(),
+        "memory_total_gb": psutil.virtual_memory().total / 10e8,
+        "memory_available_gb": psutil.virtual_memory().available / 10e8,
+    }
+
+    # Store python config
+    pyversion = sys.version_info
+    _ = {
+        "python_version": f"{pyversion.major}.{pyversion.minor}.{pyversion.micro}",
+        "aicspylibczi": aics.__version__,
+        "czifile": czifile.__version__,
+    }
+
+    # Run tests
+    #######################################################################
+
+    log.info(f"Running tests: no cluster...")
+    log.info(f"=" * 80)
+
+    all_results["no-cluster"] = _run_benchmark_suite(resources_dir=resources_dir)
+
+    #######################################################################
+
+        # Create or get log dir
+        # Do not include ms
+    log_dir_name = datetime.now().isoformat().split(".")[0]
+    log_dir = Path(f".dask_logs/{log_dir_name}").expanduser()
+    # Log dir settings
+    log_dir.mkdir(parents=True, exist_ok=True)
+
+    #######################################################################
+
+    log.info(f"Completed all tests")
+    log.info(f"=" * 80)
+
+    # Ensure save dir exists and save results
+    args.save_path.parent.mkdir(parents=True, exist_ok=True)
+    with open(args.save_path, "w") as write_out:
+        json.dump(all_results, write_out)
+
+    # # Construct and push package
+    # if args.upload:
+    #     p = Package()
+    #     p.set("results.json", args.save_path)
+    #     p.push(
+    #         "aics/benchmarks",
+    #         "s3://aics-modeling-packages-test-resources",
+    #         message=f"aics version: {aics.__version__}",
+    #     )
+
+    # Catch any exception
+# except Exception as e:
+#     log.error("=============================================")
+#     if args.debug:
+#         log.error("\n\n" + traceback.format_exc())
+#         log.error("=============================================")
+#     log.error("\n\n" + str(e) + "\n")
+#     log.error("=============================================")
+#     sys.exit(1)
```

Comment:
might as well delete all those commented out lines

Reply:
For the benchmark script, I was hoping to get some advice on how to make it more automated / in line with whatever our vision is for benchmarking on aicsimageio as I don't have a good handle on that.